### DISCUSSION TITLE
I cant find an Event in new price module that exists in old (Codeunit 7000).

### DISCUSSION THREAD

**Discussion by KristjanGaukur** [microsoft/ALAppExtensions#12290] [Q&A] [answered]
Good day.

I am using the event OnAfterGetSalesLinePrice from codeunit 7000.GetSalesLinePrice (similarly as is called from Page 47 action "ShowPrices").

I have traced the code in the new price structure (P46 -> T37 -> C7002.PickPrice) and I cannot find any events where I can subscribe to get the same information as is in the obsolete C7000.

Am I missing something that exists, or a way to get the same effect, or do I need an enhancement in the base application?

PS: Clarification: Pages such as P46 "Sales Order Subform" have the action "GetPrices" that update the price of the selected sales line. I need to interfere and add to the validation of the Sales Line based on the found Sales Price. (I have already done that when validating prices directly in the Sales Line, but need to also do this when going through the "GetPrices" action(s).)

PPS: I am guessing that I need an Event "OnAfterPickPrice" in Table 37, called from a last line in function PickPrice().

Cheers,
Kristján Gaukur Kristjánsson

**Comment by darjoo** [comment, marked as answer]
Hi Kristján,

I've reached out to the guys who implemented the new structure. Lets see what they come back with.

Edit with the response from team:
The new structure was made to repeat the old functionality and more, but the event was left out. The event however can be brought back. 

Go ahead and create an Event Request under the Issues tab and they'll get to it :)
